test/e2e/upgrade_test.go

```diff
@@ -41,6 +43,18 @@ func upgradeTestSuite(t *testing.T) {
 	require.NoError(t, err, "error testing Windows workloads")
 	defer testCtx.deleteJob(testerJob.Name)
 
+	// check BYOH enabled
+	if gc.numberOfBYOHNodes >= 1 {
+		// ensure the presence of a running workload in BYOH node
+		byohDeployment, err := testCtx.deployByohWorkload()
+		require.NoError(t, err, "error deploying BYOH workload")
```

Comment:
You're deploying a workload here, but what is actually being tested by doing so?

Reply:
Take note of https://github.com/openshift/windows-machine-config-operator/blob/7c4ed1e3950757d077ba7482f39c4af68ff25429/test/e2e/upgrade_test.go#L71-L77

Reply:
Please try and use the logic already present in the linked code and in deployWindowsWorkloadAndTester() instead of creating a new deployment.

Reply:
> You're deploying a workload here, but what is actually being tested by doing so?

As per comment: `// ensure the presence of a running workload in BYOH node`

The logic already present [deletes the deployment](https://github.com/openshift/windows-machine-config-operator/blob/master/test/e2e/upgrade_test.go#L163) before the [upgrade process is triggered](https://github.com/openshift/windows-machine-config-operator/blob/master/test/e2e/upgrade_test.go#L104), namely updating the version annotation.

So, the new workload is introduced to last during the upgrade process.

It does makes sense to accommodate the existing workload deployment [testCtx.deployWindowsWorkloadAndTester()](https://github.com/openshift/windows-machine-config-operator/blob/master/test/e2e/upgrade_test.go#L40) to persist during the upgrade.